client/src/pages/spools/show.tsx

```diff
@@ -39,6 +42,37 @@ export const SpoolShow: React.FC<IResourceComponentsProps> = () => {
     return item.price;
   };
 
+  // Function for opening an ant design modal that asks for confirmation for archiving a spool
+  const archiveSpool = async (spool: ISpool, archive: boolean) => {
+    await setSpoolArchived(spool, archive);
+    invalidate({
+      resource: "spool",
+      id: spool.id,
+      invalidates: ["list", "detail"],
+    });
+  };
+
+  const archiveSpoolPopup = async (spool: ISpool|undefined) => {
+    if (spool === undefined) {
+      return
+    }
+    // If the spool has no remaining weight, archive it immediately since it's likely not a mistake
+    if (spool.remaining_weight && spool.remaining_weight == 0) {
```

Comment:
have you tested this? remaining_weight can never be both truthy and 0 at the same time. What you want to check for the first comparison here is if it's not null I guess?

Reply:
that bit is just copied from the check done in the [table view](https://github.com/Donkie/Spoolman/blob/master/client/src/pages/spools/list.tsx#L192) so i didnt test it but yeah, it dosnt work (in either place). the result is that in both places it gives a popup confirmation even when the spool is empty. do you think it is best to just leave it as that since thats how the other button has been working or fix it in both places?

Reply:
fix it in both places would be nice :)

Reply:
fixed both, popup will now show only if the remining weight is >0g